crates/whiskers/src/grid.rs

```diff
@@ -0,0 +1,217 @@
+    pub fn build(
+        &mut self,
+        sketch: &mut Sketch,
+        callback_fn: impl FnOnce(&mut Sketch, &GridCell) + Copy,
+    ) {
+        let [module_width, module_height] = self.module_size();
+        let [gutter_width, gutter_height] = self.gutter;
+        let [columns, rows] = self.dimensions;
+        let mut cells: Vec<GridCell> = vec![];
+
+        for row in 0..rows {
+            for column in 0..columns {
+                let pos_x =
+                    self.x_y.x() + (column as f64 * module_width + column as f64 * gutter_width);
+                let pos_y =
+                    self.x_y.y() + (row as f64 * module_height + row as f64 * gutter_height);
+
+                let cell = GridCell {
+                    column,
+                    row,
+                    position: Point::new(pos_x, pos_y),
+                    size: self.module_size(),
+                };
+                callback_fn(sketch, &cell);
+                cells.push(cell);
+            }
+        }
+        self.data = cells;
+    }
+
+    /// Returns grid module's size
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    pub fn module_size(&self) -> [f64; 2] {
+        match self.size {
+            GridSize::GridBased([width, height]) => {
+                let cols = self.dimensions[0] as f64;
+                let rows = self.dimensions[1] as f64;
+                let [gutter_width, gutter_height] = self.gutter;
+
+                [
+                    (width - (cols - 1.0) * gutter_width) / cols,
+                    (height - (rows - 1.0) * gutter_height) / rows,
+                ]
+            }
+            GridSize::CellBased(s) => s,
+        }
+    }
+
+    /// Returns width of the grid
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    pub fn width(&self) -> f64 {
+        let columns = self.dimensions[0] as f64;
+        columns * self.module_size()[0] + columns * self.gutter[0]
+    }
+
+    /// Returns height of the grid
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    pub fn height(&self) -> f64 {
+        let rows = self.dimensions[0] as f64;
+        rows * self.module_size()[1] + rows * self.gutter[1]
+    }
+
+    /// Returns optional reference to a grid cell at specific column and row index
+    pub fn at(&mut self, column: usize, row: usize) -> Option<&GridCell> {
+        self.data
+            .iter()
+            .find(|cell| cell.column == column && cell.row == row)
+    }
+
+    /// Aligns grid cells to specific dimensions
+    pub fn align_to(&mut self, rect_dimensions: [f64; 2]) {
+        let [width, height] = self.module_size();
+        let diff_x = rect_dimensions[0] / width;
+        let diff_y = rect_dimensions[1] / height;
+
+        self.data.iter_mut().for_each(|cell| {
+            cell.position = Point::new(cell.position.x() + diff_x, cell.position.y() + diff_y);
+        });
+    }
```

Comment:
It was for aligning the grid to a given artboard rectangle, but since we're able to align things to the sketch with existing `sketch.center` method I think we can delete it